app/assets/javascripts/student_profile/interventions_details.js

```diff
@@ -143,6 +150,22 @@
         <p>If your data fits into one of these categories, it's a Service. Otherwise, it's a Note.</p>"
     },
 
+    renderRestrictedNotesButtonIfAppropriate: function(){
+      var self = this;
+
+      if (this.props.currentEducator.can_view_restricted_notes){
+        return dom.button({
+          className: 'btn',
+          style: styles.restrictedNotesButton,
+          onClick: function(){
+            location.href = '/students/' + self.props.student.id + '/restricted_notes';
```

Comment:
I think you can avoid navigating to another page, and save yourself a lot of repetitive code, by making whether or not you are viewing restricted notes part of the state of the `PageContainer` instead.